### PR TITLE
add bottom navigation view in session detail screen

### LINKED ISSUE #164: add bottom navigation view in session detail screen
**Idea Description**

- share
- scroll to top
- favorite
  - when tapped set favorite button toggled

**Reference images and links**

Design Kit
https://www.figma.com/file/XsVzpDZSkEQANgCZLRTDDy/DroidKaigi-2022-Material-3-Design-Kit?node-id=11%3A1833
App Design
https://www.figma.com/file/NcSMs6dMsD88d4wOY0g3rK/DroidKaigi-2022-Conference-App?node-id=0%3A1

### PR DESCRIPTION
**Pull request by KoutaMatsushita**
## Issue
- close #164

## Overview (Required)
- Add session detail bottom navigation actions
  - Share
    - call platform share feature
  -  Go to the floor map screen
    - floor map screen not implement yet....
    - For now, implemented a mock
  - Register the schedule to Google calendar
    - call platform share feature

## Links
- [design](https://www.figma.com/file/NcSMs6dMsD88d4wOY0g3rK/DroidKaigi-2022-Conference-App?node-id=21%3A583)

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

File: app-android/src/main/java/io/github/droidkaigi/confsched2022/impl/AndroidShareManager.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import android.content.Context
+import androidx.core.app.ShareCompat
+import io.github.droidkaigi.confsched2022.ui.ShareManager
+
+class AndroidShareManager(private val context: Context) : ShareManager {
+    override fun share(text: String) {
+        ShareCompat.IntentBuilder(context)
+            .setText(text)
+            .setType("text/plain")
+            .startChooser()
```
Comment (RyuNen344) [inline app-android/src/main/java/io/github/droidkaigi/confsched2022/impl/AndroidShareManager.kt:12]: I don't think there is one, but may I ask you to add handling when `startChooser` throws exception as well?

File: core-ui/src/main/java/io/github/droidkaigi/confsched2022/ui/CalendarRegistration.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+     */
+    fun register(
+        startsAtMilliSeconds: Long,
+        endsAtMilliSeconds: Long,
+        title: String,
+        location: String,
+    )
+}
+
+val LocalCalendarRegistration = staticCompositionLocalOf<CalendarRegistration> {
```
Comment (RyuNen344) [inline core-ui/src/main/java/io/github/droidkaigi/confsched2022/ui/CalendarRegistration.kt:21]: lovely ❤️

File: app-android/src/main/java/io/github/droidkaigi/confsched2022/impl/AndroidShareManager.kt (hunk 1 above)
Comment (KoutaMatsushita) [reply app-android/src/main/java/io/github/droidkaigi/confsched2022/impl/AndroidShareManager.kt]: Added! 🚀

**Review by RyuNen344 — APPROVED** [review]
LGTM 💯

**Pull request merged**